src/EditorFeatures/CSharpTest/InitializeParameter/InitializeMemberFromParameterTests.cs

```diff
@@ -0,0 +1,573 @@
+        {
+            await TestInRegularAndScript1Async(
+@"
+class C
+{
+    private string s;
+
+    public C([||]string s)
+    {
+    }
+}",
+@"
+class C
+{
+    private string s;
+
+    public C(string s)
+    {
+        this.s = s;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInitializeParameter)]
+        public async Task TestEndOfParameter1()
+        {
+            await TestInRegularAndScript1Async(
+@"
+class C
+{
+    private string s;
+
+    public C(string s[||])
+    {
+    }
+}",
+@"
+class C
+{
+    private string s;
+
+    public C(string s)
+    {
+        this.s = s;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInitializeParameter)]
+        public async Task TestEndOfParameter2()
+        {
+            await TestInRegularAndScript1Async(
+@"
+class C
+{
+    private string s;
+
+    public C(string s[||], string t)
+    {
+    }
+}",
+@"
+class C
+{
+    private string s;
+
+    public C(string s, string t)
+    {
+        this.s = s;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInitializeParameter)]
+        public async Task TestInitializeFieldWithUnderscoreName()
+        {
+            await TestInRegularAndScript1Async(
+@"
+class C
+{
+    private string _s;
+
+    public C([||]string s)
+    {
+    }
+}",
+@"
+class C
+{
+    private string _s;
+
+    public C(string s)
+    {
+        _s = s;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInitializeParameter)]
+        public async Task TestInitializeWritableProperty()
+        {
+            await TestInRegularAndScript1Async(
+@"
+class C
+{
+    private string S { get; }
+
+    public C([||]string s)
+    {
+    }
+}",
+@"
+class C
+{
+    private string S { get; }
+
+    public C(string s)
+    {
+        S = s;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInitializeParameter)]
+        public async Task TestInitializeFieldWithDifferentName()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C
+{
+    private string t;
+
+    public C([||]string s)
+    {
+    }
+}",
+@"
+class C
+{
+    private string t;
+
+    public C(string s)
+    {
+        S = s;
+    }
+
+    public string S { get; }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInitializeParameter)]
+        public async Task TestInitializeNonWritableProperty()
+        {
+            await TestInRegularAndScript1Async(
+@"
+class C
+{
+    private string S => null;
+
+    public C([||]string s)
+    {
+    }
+}",
+@"
+class C
+{
+    private string S => null;
+    public string S1 { get; }
+
+    public C(string s)
+    {
+        S1 = s;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInitializeParameter)]
+        public async Task TestInitializePropertyWithBadName()
```

Comment:
I'm just validating that we don't hook up to a property with a name that certainly doesn't match.  will rename test.